com.unity.netcode.gameobjects/Runtime/Serialization/NetworkBehaviourReference.cs

```diff
@@ -0,0 +1,103 @@
+using System;
+using System.Runtime.CompilerServices;
+
+namespace Unity.Netcode
+{
+    /// <summary>
+    /// A helper struct for serializing <see cref="NetworkBehaviour"/>s over the network. Can be used in RPCs and <see cref="NetworkVariable{T}"/>.
+    /// Note: network ids get recycled by the NetworkManager after a while. So a reference pointing to
+    /// </summary>
+    public struct NetworkBehaviourReference : INetworkSerializable, IEquatable<NetworkBehaviourReference>
```

Comment:
wrt:

> The fix for this would be to cache the NetworkObject inside the reference after resolving it for the first time but this does not work with NetworkVariable as the struct would no longer be unmanaged.

I think this being a struct, value type and let's say documented with its proper usage etc would be good enough IMHO.
Maybe exposing internal `m_NetworkBehaviourId` over a public get-only ulong property would give more clues about this thing is actually working inside (+ maybe another TryGet API that accepts that ID too?).
To be fair, I'd like people to use this like: `MyServerRpc(new NetworkBehaviourReference(MyNetworkBehaviour));` etc (you get the idea).

long story short, I wouldn't worry about it too much :P

Reply:
also what about super-generic `GameObject` one where we try to get `NetworkObject` or `NetworkBehaviour` component from it to serialize over the network? 👀

Reply:
You can actually do `MyServerRpc(MyNetworkBehaviour)` and the implicit operator will make it work automagically as long as the parameter on the RPC function itself uses the reference.

Regarding the GameObject one. I think taking a `NetworkBehaviour` from it would be a bit too far fetched so I'm afraid people might get confused as to why they can send some `GameObjectReference`s and not others. `NetworkObjectReference` also has implicit operators from/to `GameObject` so you can pass an GameObject into an RPC with a `NetworkObjectReference` parameter.

Reply:
oh this implicit operator, just saw this :)
```cs
public static implicit operator NetworkObjectReference(GameObject gameObject) => new NetworkObjectReference(gameObject);
```

Reply:
I agree with your second point too!

Reply:
actually, what about adding a few other RPCs for these implicit operators into tests you implemented in this PR?
that way, they'll be tested + will be good sample code as well — what do you think?